unit_test/tstUniformCartesianMeshMapping.hpp

```diff
@@ -0,0 +1,347 @@
+/****************************************************************************
+ * Copyright (c) 2021 by the Picasso authors                                *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Picasso library. Picasso is distributed under a *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Picasso_BatchedLinearAlgebra.hpp>
+#include <Picasso_CurvilinearMesh.hpp>
+#include <Picasso_Types.hpp>
+#include <Picasso_UniformCartesianMeshMapping.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cmath>
+
+#include <gtest/gtest.h>
+
+using namespace Picasso;
+
+namespace Test
+{
+//---------------------------------------------------------------------------//
+void mappingTest3d()
+{
+    // Global parameters.
+    Kokkos::Array<double, 6> global_box = { -10.0, -10.0, -10.0,
+                                            10.0,  10.0,  10.0 };
+    double cell_size = 0.5;
+    int num_cell = 20.0 / cell_size;
+    int halo_width = 1;
+    Kokkos::Array<bool, 3> periodic = { true, false, true };
+
+    // Partition only in the x direction.
+    std::array<int, 3> ranks_per_dim = { 1, 1, 1 };
```

Comment:
if partitioned only in x-dir, then ranks_per_dim[0] > 1?

Reply:
can be `>=`